CHANGES.rst

```diff
@@ -11,6 +11,8 @@ Features / Changes
 ~~~~~~~~~~~~~~~~~~~~~
 * Explicitly require ``MAGPIE_ADMIN_USER`` and ``MAGPIE_ADMIN_PASSWORD`` to be updated through configuration and
   application restart. Update is forbidden through the API and UI.
+* Add UI loading animation while sync operation is in progress to indicate some user feedback that it was registered
+  and is running until completion as it can take a while to parse all remote resources (depends on amount and latency).
```

Comment:
"it can take a while to parse all remote resources (depends on amount and latency)" this worries me a bit here.

How does Magpie "scan" for Thredds data structure?  Crawling via Thredds web UI?

In production we have a massive dataset on Thredds.  Last time we triggered the pavics crawler, it took 2 days to finish crawling Thredds !

What would happen if somehow Thredds and/or Mapie is restarted in the middle of the scan?

Does Magpie still operating "normally" performance wise during the scan?

Looks like Magpie also crawl the wpsoutput folder, that is also a source of massive content that is more or less useless.  Is there a way to exclude it from the scan (as a config option, not hardcoded).  Will surely help the scan to finish faster.

Reply:
For the case of THREDDS, it uses `threddsclient.read_url` recursively for each children directories returned by each previous response. There is a `max_depth` parameter to limit this process. The idea is to go as far as needed to generate the structure required to apply permissions to avoid registering each resource by hand, but specific definitions can still be added manually after. If permissions are all applied to (at most) the displayed 4 levels, then no more pointless crawling should be done. For the moment, there is only `max_depth` (hardcoded) without any additional exclude, but both of these could be added to the thredds JSON config. 

Normally, one worker should be processing the sync, while others still can receive and answer to requests. 
If THREDDS stopped midway, I think the sync will fail because of error response. 
If Magpie is stops, it will simply not complete the scan.

Reply:
Next time Thredds/Magpie comes back on line, will Magpie resume the scan?